tests/mysql/test-mysql.h

```diff
@@ -112,6 +112,13 @@ class test_context : public test_context_base
         // SQL mode to avoid silent truncation of too long values.
         return sql_mode.find("STRICT_") == std::string::npos;
     }
+
+    virtual bool has_trailing_spaces_in_char_type(session& sql) const 
+    {
+        //turn on padding on mysql - trimming by default
+        sql << "SET @@session.sql_mode = 'PAD_CHAR_TO_FULL_LENGTH'";
```

Comment:
According to the docs, this mode is new in 5.1.20, I don't know if it there is still any real possibility of encountering an older MySQL version out there, but if we can check for the error here, it would be worth doing it.